src/main/groovy/com/miniasaaslw/repository/payment/PaymentRepository.groovy

```diff
@@ -9,6 +9,14 @@ class PaymentRepository implements BaseEntityRepository {
     public static DetachedCriteria<Payment> query(Map search) {
         DetachedCriteria<Payment> query = Payment.where(defaultQuery(search))
 
+        query = query.where {
+            if (search.containsKey("payerName[like]")) {
+                payer {
```

Comment:
aqui o certo é usar createAlias quando precisa de join

e condicionar, fazer o join somente quando o filtro payerName[like] for passado

Reply:
https://github.com/orgs/L-W-payments/projects/1/views/1?pane=issue&itemId=65033648